module/zcommon/zfs_prop.c

```diff
@@ -574,6 +673,10 @@ zfs_prop_init(void)
 	    PROP_TYPE_NUMBER, PROP_READONLY, ZFS_TYPE_FILESYSTEM,
 	    "PRIV_PROP");
 
+	zprop_register_impl(ZFS_PROP_ZSTD_LEVEL, "zstd_compress_level",
+	    PROP_TYPE_NUMBER, ZIO_ZSTDLVL_DEFAULT, NULL, PROP_INHERIT,
+	    ZFS_TYPE_DATASET, "<zio_zstd_levels>", "ZSTDLEVEL", B_TRUE,
+	    B_FALSE, NULL);
```

Comment:
i dont think so. the level is removed from compression settings and moved to this attribute within the code. but correct me if i'm wrong

Reply:
My concern is that unless it's set readonly you may be able to run `zfs set zstd_compress_level=<num> dataset`, which I don't think we want to allow.  Can you check this behavior.

Reply:
i checked this  and it works. is changing the level with a direct property a problem? i mean its usually even better than the gzip idea with different algorithm names just to describe different levels. this looks more like a hack than the zstd_compress_level variant

Reply:
i dont really have a oppinion here. from user perspective setting the level makes sense, but i have the same oppinion for gzip levels here. so i will hide it with next commit once someone made a decision how to handle it in future for all compression algorithms in the same way.

Reply:
it seems that PROP_READONLY cannot be used since PROP_INHERIT is important here, but cannot coexist with the readonly property. so this is some sort of api limitation i cannot work around right now

Reply:
it seems that PROP_READONLY cannot be used since PROP_INHERIT is important here, but cannot coexist with the readonly property. so this is some sort of api limitation i cannot work around right now

Reply:
i cannot use readonly here. the inherit option is important but cannot coexist with readonly

Reply:
i cannot use readonly here. the inherit option is important but cannot coexist with readonly

Reply:
i cannot use readonly here. the inherit option is important but cannot coexist with readonly

Reply:
OK, we definitely need to keep `PROP_INHERIT`, it's unfortunate we can't set both.  Adding a case to `zfs_check_settable` for the hidden property is probably the way handle this.  Arguably we need one any to verify the feature is enabled.

Reply:
or we change the api a little bit to set readonly as masked operator. hidden properties doesnt mean they can't be changed. sometimes people might want to keep it hidden for debugging purpose. consider that even if this property here is hidden, its still harmless. it lets allow you to change the compression strength, but wont break anything. the classic compression setting still works anyway